compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1881,14 +1880,19 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
     // The length of the returned iterator is nearly always 0 or 1 and this
     // method is fairly hot.
     fn impl_or_trait_item(&self, def_id: DefId) -> SmallVec<[ty::AssocItem; 1]> {
+        let relevant_kind_for_mode = |kind| match (self.mode, kind) {
+            (Mode::MethodCall, ty::AssocKind::Fn) => true,
+            (Mode::Path, ty::AssocKind::Const | ty::AssocKind::Fn) => true,
+            _ => false,
+        };
```

Comment:
I kinda like this approach, but can you move this to be an associated method rather than a closure?

Reply:
Done